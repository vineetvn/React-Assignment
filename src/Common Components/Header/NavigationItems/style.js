import styled from 'styled-components';
import {FaSearch} from 'react-icons/fa';
import {MdLocalGroceryStore} from 'react-icons/md';
import {CgProfile} from 'react-icons/cg';

const Ul = styled.ul`
    margin: 0;
    padding: 0;
    padding-right: 100px;
    list-style: none;
    flex: 80%;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    height:100%;
    font-family: ${props => (props.theme.fontFamily.Montserrat)};
    font-weight: ${props => (props.theme.fontWeight.bold)};

    @media only screen and (max-width: 900px) {
        margin: 20px;
        padding-right: 0;
    }

`

export const SearchIcon = styled(FaSearch) `
    font-size: ${props => (props.theme.fontSize.small)};
    line-height: ${props => (props.theme.fontSize.small)};
`
export const StoreIcon = styled(MdLocalGroceryStore) `
    font-size: ${props => (props.theme.fontSize.small)};
    line-height: ${props => (props.theme.fontSize.small)};
`

export const Menu = styled(CgProfile) `
    font-size: ${props => (props.theme.fontSize.medium)};
    line-height: ${props => (props.theme.fontSize.small)};
    display: flex;
`

export const Div = styled.div `
    position: relative;
    margin-left: 20px;
    cursor: pointer;
    line-height: 27px;
`

export const Badge = styled.span `
    position: absolute;
    top: -13px;
    right: -15px;
    width: 25px;
    height: 25px;
    padding: 0 0 10 0px;
    border-radius: 50%;
    background-color: ${props => props.color};
    color: ${props => (props.theme.color.white)};
    text-align: center;
`
export {Ul};
