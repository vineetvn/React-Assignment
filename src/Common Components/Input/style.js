import styled from 'styled-components';

const Inputelement = styled.input `
    padding: 12px 18px;
    line-height: ${props => (props.theme.lineHeight.medium)};
    font-size: ${props => (props.theme.fontSize.mall)};
    font-weight: ${props => (props.theme.fontWeight.medium)};
    font-family: ${props => (props.theme.fontFamily.Lato)};

`
const Textarea = styled.textarea `
    height: 200px;
    padding: 12px 18px;
`

const Div = styled.div `
    max-width: 100%;
    flex: 0.5;
    display: flex;
    flex-direction: column;
    margin: 0 20px 20px 20px;
    box-sizing: border-box;

    @media only screen and (max-width: 900px) {
        margin: 20px 0;
    }
`

const P = styled.p `
    color: ${props => (props.theme.color.red)};

`

export {Inputelement, Div, Textarea, P};