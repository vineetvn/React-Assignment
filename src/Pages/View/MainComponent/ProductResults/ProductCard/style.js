import styled from 'styled-components';
import {HiShoppingCart} from 'react-icons/hi'

export const Div = styled.div `
    margin: 15px;
    display: flex;
    flex-direction: column;
    height: 320px;
    width: 250px;
    border: 1px solid ${props => props.theme.color.silverGrey};
    font-family: ${props => props.theme.fontFamily.Montserrat};
    box-sizing: border-box;
    text-align: center;
    position: relative;

    @media only screen and (max-width: 1000px) {
        height: 320px;
        width: 180px;
    } 

    @media only screen and (max-width: 685px) {
        height: 300px;
        width: 70%;
        margin: 15px auto;
    } 

    @media only screen and (max-width: 450px) {
        height: 300px;
        width: 90%;
        margin: 15px auto;
    }
`

export const ShoppingCart = styled(HiShoppingCart) `
    color: white;
    background-color: black;
    padding: 10px;
    position: absolute;
    right: 0;
    top: 190px;
`
export const Img = styled.img `
    height: 225px;
    width: 100%;
    display: block;
`

export const ImageDetails = styled.div `
    height: 50px;
    box-sizing: border-box;
    font-family: ${props => props.theme.fontFamily.Montserrat}
`

export const H4 = styled.h4 `
    margin: 15px;
    text-align: center;
    box-sizing: border-box;
`

export const Price = styled.span `
    margin: 15px;
    text-align: center;
    box-sizing: border-box;
    margin-left: 5px;
`

export const OfferPrice = styled.span `
    margin: 15px;
    text-decoration: line-through;
    text-align: center;
    box-sizing: border-box;
`