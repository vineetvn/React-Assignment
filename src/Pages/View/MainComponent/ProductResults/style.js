import styled from 'styled-components';

export const OuterDiv = styled.div `
    flex: 70%;
    width: 100%;
    height: auto;
    display: flex;

    @media only screen and (max-width: 500px) {
        flex: 100%;
    }
`

export const MainSection = styled.div `
    flex: 75%;
`

export const Range = styled.div `
    margin: 15px 0;
    width: 100%;
    font-family: ${props => props.theme.fontFamily.Montserrat};
    display: flex;
    @media only screen and (max-width: 1000px) {
        flex-direction: column;
    }
`

export const P = styled.p `
    margin: 50px;
    font-family: ${props => props.theme.fontFamily.Montserrat};
`

export const Filter = styled.div `
    margin: 0 20px;
    @media only screen and (max-width: 1000px) {
        margin: 25px 0;
    }

`

export const Div = styled.div `
    height: auto;
    display: flex;
    flex-wrap: wrap;
    margin: 20px;

    @media only screen and (max-width: 500px) {
        flex-wrap: nowrap;
        flex-direction: column;
    }
`

export const Div1 = styled.div `
    width: 100%;
`

export const Span = styled.span `
    margin: 0 20px 0 40px;

    @media only screen and (max-width: 700px) {
        margin: 0 40px;
    }
    @media only screen and (max-width: 350px) {
        display: none;
    }
`

export const Input = styled.input `
    -webkit-appearance: none;
    background: grey; 
    outline: none;
    height: 5px;
    margin: 0 20px;
    &::-webkit-slider-thumb {
        -webkit-appearance: none;
        width: 15px;
        height: 15px;
        background: black;
        cursor: pointer;
    }

    @media only screen and (max-width: 700px) {
        margin: 0 15px;
    }

    @media only screen and (max-width: 350px) {
        margin: 0 30px;
    }

`

export const Search = styled.input `
    border: 1px solid black;
    height: 20px;
    font-family: ${props => props.theme.fontFamily.Lato};

    @media only screen and (max-width: 700px) {
        margin: 0 0 0 15px;
    }

    @media only screen and (max-width: 350px) {
        margin: 0 30px;
    }

`