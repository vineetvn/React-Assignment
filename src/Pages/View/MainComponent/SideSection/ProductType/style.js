import styled from 'styled-components';
import {Link} from 'react-router-dom';

export const Div = styled.div `
    flex:25%auto;
    margin: 130px 50px 0 50px;
    display: flex;
    flex-direction: column;
    text-align: center;
    background-color: ${props => props.theme.color.silverGrey};
    color: ${props => props.theme.color.darkCharcoal};
    font-size: ${props => props.theme.fontSize.productName};
    font-family: ${props => props.theme.fontFamily.Montserrat};
    padding: 30px 10px 10px 10px;

    @media only screen and (max-width: 1000px) {
        margin: 25px;
        flex: 25%;

    }
`

export const StyledLink = styled(Link) `
    text-decoration: none;
    color: ${props => props.theme.color.red};
    font-weight: ${props => props.theme.fontWeight.bold};
    padding-bottom: 20px;
`