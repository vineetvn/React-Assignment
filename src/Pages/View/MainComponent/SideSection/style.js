import styled from 'styled-components';

import Woman from '../../../../Assets/Images/Woman.jpg';

export const ImageWrapper = styled.div `
    background: url('${Woman}');
    flex: 75%;
    background-repeat: no-repeat;
    background-position: center;
    height: 100%auto;
    margin: 50px;

    @media only screen and (max-width: 1000px) {
        margin: 25px;
        flex: 75%;
        height: 100%;
    }    
`

export const Div = styled.div `
    display: flex;
    flex-direction: column;
    flex: 30%;

    @media only screen and (max-width: 1000px) {
        margin-top: 100px;
    } 
`