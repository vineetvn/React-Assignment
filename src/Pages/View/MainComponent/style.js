import styled from 'styled-components';

export const Div = styled.div `
    display: flex;
    height: auto;
    width: 100%;

    @media only screen and (max-width: 600px) {
        flex-direction: column;
    }
`