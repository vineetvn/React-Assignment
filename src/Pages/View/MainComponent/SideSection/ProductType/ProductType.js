import React from 'react'
import { Div, StyledLink } from './style'
import {connect} from 'react-redux'

function ProductType(props) {
    let name;
    props.relatedProducts.filter(prod => {
        if(prod.id === 0) name = prod.name;
        return 0;
    })
    return (
        <Div>
            <h2>Welcome To The Shop Section</h2>
            {props.locationState + 1 ? <h3>{name} Section</h3> : null}
            <p>Return back to  <StyledLink to='/home'>Home</StyledLink></p>
        </Div>
    )
}

const mapStateToProps = state => {
    return {
        relatedProducts: state.view.relatedProducts,
        locationState: state.view.locationState
    }
}

export default connect(mapStateToProps, null)(ProductType)
