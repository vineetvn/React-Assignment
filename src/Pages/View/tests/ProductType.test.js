import Adapter from 'enzyme-adapter-react-16';
import {configure} from 'enzyme';
import { shallowRender } from '../../../testUtils';
import {ProductType} from '../MainComponent/SideSection/ProductType/ProductType';
import { StyledLink } from '../MainComponent/SideSection/ProductType/style';

configure({adapter: new Adapter()});

describe('ProductType', () => {
    let wrapper;
    let props = {
        relatedProducts: [
            {
                id: 0,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            }
        ],
        locationState: 0
    }

    beforeEach(() => {
        wrapper = shallowRender(ProductType, false, props)
    })

    it('should render the component', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('should ssimulate the link', () => {
        expect(wrapper.find(StyledLink).simulate('click'))
    })
});