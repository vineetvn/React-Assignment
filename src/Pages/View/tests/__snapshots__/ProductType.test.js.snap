// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductType should render the component 1`] = `ShallowWrapper {}`;
